test/getGrist.ts

```diff
@@ -196,19 +223,27 @@ export class GristUtils extends GristWebDriverUtils {
     await this.waitForServer();
   }
 
-  public async setCustomWidgetAccess(option: "none"|"read table"|"full") {
+  public async setCustomWidgetAccess(option: "none" | "read table" | "full") {
     const text = {
-      "none" : "No document access",
+      "none": "No document access",
       "read table": "Read selected table",
       "full": "Full document access"
     };
     await this.driver.find(`.test-config-widget-access .test-select-open`).click();
     await this.driver.findContent(`.test-select-menu li`, text[option]).click();
   }
 
-  public async setCustomWidgetMapping(name: string, value: string|RegExp) {
-    const click = (selector: string) => driver.findWait(`${selector}`, 2000).click();
-    const toggleDrop = (selector: string) => click(`${selector} .test-select-open`);
+  public async setCustomWidgetMapping(name: string, value: string | RegExp) {
+    const click = async (selector: string) => {
+      try {
+        await driver.findWait(selector, 2000).click();
+      } catch (e) {
+        //sometimes here we get into "detached" state and test fail.
+        //if this happened, just try one more time
+       await  driver.findWait(selector, 2000).click();
```

Comment:
```suggestion
        await driver.findWait(selector, 2000).click();
```

Reply:
Did this change turn out to be good enough for the tests to be reliable?

Reply:
i did some empirical tests - before in average one in ten tests run was failing on my machine on a set of 50 runs. After this change i get no single failure in 80 runs.